### PR TITLE
1.2 - Added Sidebar with OnClick Functionality

### PR DESCRIPTION
**Pull request by YannickLa**
Added a sidebar, where clicking on a button will print dynamic text in the console.

By: @eva128 @YannickLa

### CODE REVIEW

File: Unity/Assets/SidebarButtonHandler.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+using UnityEngine;
+using UnityEngine.UI;
+
+public class SidebarButtonHandler : MonoBehaviour
+{
+    public Button sidebarButton;
+
+    void Start()
+    {
+        Button btn = sidebarButton.GetComponent<Button>();
```
Comment (ArminT28) [inline Unity/Assets/SidebarButtonHandler.cs:12]: Is this code connected to a button in the Scene?

File: Unity/Assets/SidebarButtonHandler.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        string btnText = btn.name;
+        btn.onClick.AddListener(() => TaskOnClick(btnText));
```
Comment (ArminT28) [inline Unity/Assets/SidebarButtonHandler.cs:14]: To which button are you adding the Listener exactly?

File: Unity/Assets/SidebarButtonHandler.cs (hunk 1 above)
Comment (YannickLa) [reply Unity/Assets/SidebarButtonHandler.cs:12]: Yes, it is connected to every button in the sidebar and will then dynamically show, which button has been pressed in the console.

Comment (tropper26) [reply Unity/Assets/SidebarButtonHandler.cs:12]: Can you do a clean clone of the branch you've pushed to see how it works on our end? I think its missing a file and that's why its not working

Comment (YannickLa) [reply Unity/Assets/SidebarButtonHandler.cs:12]: Sure, let me try

Comment (YannickLa) [reply Unity/Assets/SidebarButtonHandler.cs:12]: Apparently the script hadn't saved, when I assigned it to the buttons. I reassigned it, closed and opened Unity again. It worked this time. Can you check, please?

Comment (tropper26) [reply Unity/Assets/SidebarButtonHandler.cs:12]: image

im not sure what's wrong with it, if I add the script again then it works

Comment (tropper26) [reply Unity/Assets/SidebarButtonHandler.cs:12]: I guess we'll check on monday

**Review by tropper26 — APPROVED** [review]

**Pull request merged**